### ISSUE TITLE
Nullable parameters for custom types

### ISSUE THREAD

**Issue by BernieWhite** [Azure/PSRule.Rules.Azure#2489]
### Description of the issue

Similar to #2488 this issue occurs for custom types.

When nullable parameters are defined in a module and the consuming module does not pass through a value (resulting in a null) an expansion error is raised.

### Error messages

`The parameter named 'corsRules' was not set or a defaultValue was defined.`

### Reproduction

```bicep

param minTLSVersion string?
param corsRules corsRule

type corsRule = {
  allowedHeaders: string[]
  allowedMethods: string[]
  allowedOrigins: string[]
  exposedHeaders: string[]
  maxAgeInSeconds: int
}[]?

resource storage 'Microsoft.Storage/storageAccounts@2023-01-01' = {
  name: 'test'
  #disable-next-line no-loc-expr-outside-params
  location: resourceGroup().location
  sku: {
    name: 'Standard_LRS'
  }
  kind: 'StorageV2'
  properties: {
    minimumTlsVersion: minTLSVersion ?? 'TLS1_2'
  }

  resource blob 'blobServices' = {
    name: 'default'
    properties: {
      cors: {
        corsRules: corsRules ?? []
      }
    }
  }
}
```

### Version of PSRule

2.9.0

### Version of PSRule for Azure

1.30.2

### Additional context

Raised by @AlexanderSehr microsoft/PSRule#1640
